### ISSUE TITLE
Question about Gradient Accumulation

### ISSUE THREAD

**Issue by Six-Bit-TX** [amap-cvlab/CLoD-GS#2]
```
            if not hasattr(opt, 'batch_mode') or opt.batch_mode == "per_scale":
                # Original mode: backward pass for each scale separately
                loss.backward()
            else:
                # Accumulate loss for all scales and do a single backward pass
                if i < len(scales) - 1:
                    # Accumulate loss without backward
                    accumulated_loss += loss
                else:
                    # On the last scale, do the backward pass with accumulated loss
                    if accumulated_loss > 0:
                        # Add the final loss to accumulated loss
                        accumulated_loss += loss
                        accumulated_loss.backward()
                    else:
                        # If there's only one scale or accumulated_loss is still 0
                        loss.backward()
```
Aren't these two modes the same? 

From a mathematical/gradient perspective, these two approaches are actually equivalent! Here's why:
If you have losses from multiple scales (L₁, L₂, L₃):
Mode 1: L₁.backward(), then L₂.backward(), then L₃.backward()
Mode 2: (L₁ + L₂ + L₃).backward()
Since PyTorch accumulates gradients by default (gradients are additive), both produce the same final gradients:
∇(L₁ + L₂ + L₃) = ∇L₁ + ∇L₂ + ∇L₃

**Comment by czg-first** [comment]
From a mathematical perspective, the two modes are indeed equivalent. However, in practical engineering, we found that the per_scale mode (default) performs more stably in practice. And the accumulated loss mode is theoretically cleaner but shows no actual improvement in training results.

Although mathematically equivalent, factors such as framework implementation details, memory management, and numerical precision may lead to minor differences in practice. Therefore, we recommend using the thoroughly validated per_scale mode.
